aios/trajectory.py

```diff
@@ -0,0 +1,186 @@
+        )
+
+        opti.subject_to(
+            -1 * math.pi / 2 < self.phi + deltaPhi 
+        )
+
+        opti.subject_to(
+            self.phi + deltaPhi < math.pi / 2
+        )
+
+        opti.subject_to(
+            self.MAX_HEIGHT - (self.NOTE_THICKNESS / 2) >
+           ( (self.v + deltaV) * casadi.sin(deltaTheta + self.theta) * (self.y - self.OPENING_GAP)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy ) )
+           - 0.5 * self.GRAVITATIONAL_ACCELERATION * ((self.y - self.OPENING_GAP)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )) ** 2
+        )
+
+        opti.subject_to(
+            self.MIN_HEIGHT + (self.NOTE_THICKNESS / 2) >
+           ( (self.v + deltaV) * casadi.sin(deltaTheta + self.theta) * (self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy ) )
+           - 0.5 * self.GRAVITATIONAL_ACCELERATION * ((self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )) ** 2
+        )
+
+        opti.subject_to(
+            0 <=
+           ( (self.v + deltaV) * casadi.sin(deltaTheta + self.theta) * (1)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy ) )
+           - self.GRAVITATIONAL_ACCELERATION * (self.y)/ ( ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )) ** 2
+        )
+
+        opti.subject_to(
+            (-1 * self.SPEAKER_WIDTH + self.NOTE_OUTER_RADIUS) / 2 < self.x + (( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.sin(self.phi + deltaPhi)) + self.vx) * (self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )
+        )
+
+        opti.subject_to(
+            (self.SPEAKER_WIDTH - self.NOTE_OUTER_RADIUS) / 2 > self.x + (( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.sin(self.phi + deltaPhi)) + self.vx) * (self.y)/ ( ( (self.v + deltaV) * casadi.cos(self.theta + deltaTheta) * casadi.cos(self.phi + deltaPhi) ) + self.vy )
+        )
+
+
+        
+        opti.solver('ipopt')
+        try:
+            solutions = opti.solve()
+        except:
+            return ValueError("No calculations for shooting available at this position.")
+
+
+        return {
+            'speed' : round(solutions.value(deltaV) + self.v, 3), 
+            'angle' : round(solutions.value(deltaTheta) + self.theta, 3),
+            'heading' : round(solutions.value(deltaPhi) + self.phi , 3)
+        }
+
+    def canShootWhileStationary(self) -> bool:
+        if (self.y < (self.OPENING_GAP / self.SPEAKER_WIDTH ) * (self.x) - (self.OPENING_GAP / 2)) and (self.y < -1 * (self.OPENING_GAP / self.SPEAKER_WIDTH ) * (self.x) - (self.OPENING_GAP / 2)):
```

Comment:
```suggestion
        return (self.y < (self.OPENING_GAP / self.SPEAKER_WIDTH ) * (self.x) - (self.OPENING_GAP / 2)) and (self.y < -1 * (self.OPENING_GAP / self.SPEAKER_WIDTH ) * (self.x) - (self.OPENING_GAP / 2))
```

Reply:
i know you said this was giving you an error, but it really shouldn't be, so can you try it again and if it doesn't work let me know what error it's giving you and we'll try to figure it out